lib/internal/loader/Translators.js

```diff
@@ -55,10 +55,13 @@ translators.set('commonjs', async (url) => {
 // through normal resolution
 translators.set('builtin', async (url) => {
   debug(`Translating BuiltinModule ${url}`);
-  return createDynamicModule(['default'], url, (reflect) => {
+  const id = url.substr(5);
+  const module = NativeModule.requireNamespace(id);
+  const keys = Object.keys(module);
+  return createDynamicModule(keys, url, (reflect) => {
     debug(`Loading BuiltinModule ${url}`);
-    const exports = NativeModule.require(url.slice(5));
-    reflect.exports.default.set(exports);
+    for (const key of keys)
+      reflect.exports[key].set(module[key]);
```

Comment:
I'm wondering if we could make the `requireNamespace` module return a `dynamic module` already so we avoid this kind of double-interation?

Reply:
dynamic modules are created asynchronously and there's a bit of overhead for them anyway. furthermore i'd like to keep as much module stuff out of cjs stuff as possible.

Reply:
But you follow the benefit of being able to keep setters in sync? Thoughts on the setter filtering?

Reply:
wasn't the general idea that setters would be retired anyway?

Reply:
If we can make them work with backwards compatibility, while providing a forward API (`setX`) for ES modules, it gives us a healthier deprecation path instead of rushing users to a deadline. But if we can get those deprecations going too then great. It's a process though so nothing wrong with tackling it from multiple directions.

Reply:
looking into this, i would have to replace the setter on the public api, which is mutable and could be overwritten, breaking the link. i don't think it can be tackled from this direction

Reply:
@devsnek the idea would be to support practical use cases of setters, not cases where "setting" happens to work but is not endorsed. Eg https://github.com/nodejs/node/blob/master/lib/domain.js#L43 could be supported if that code itself could get a handle on the ES namespace and update it at the same time.

Just exploring these boundaries here, only suggesting not filtering setters for now to support eg `import { domain } from 'process'` and that there is a path to keep setters in sync if ES namespaces can be moved down a level of abstraction.

Reply:
Re sync / async boundary - actual execution of module is sync so getters will be in sync with setters at that point. Once execution is in place, if handle from CJS side can be updated sync on ES namespace execution, handle updates on set would remain in sync too. So we get full parity there fine despite async.

Reply:
like jasnell is doing below, i think it would be better to have process.{set,get}Domain

Reply:
Definitely, if we can do this for every case then great, but will believe it when I see it :)